lib/redis/distributed.rb

```diff
@@ -694,6 +694,11 @@ def zmscore(key, *members)
       node_for(key).zmscore(key, *members)
     end
 
+    # Iterate over keys, removing members from the first non empty sorted set found.
+    def zmpop(*keys, modifier: "MIN", count: nil)
+      keys.find { |key| node_for(key).zmpop(key, modifier: modifier, count: count) }
```

Comment:
I'm not sure if this is right, it seems to make sense? The keys are on different nodes, so we can't really use zmpop otherwise we're assuming they're on the same node right? Would it be better to just raise not implemented?

Reply:
I'd say do what `zpopmin` and `zpopmax` do, which is to call `ensure_same_node`.